src/cat-data-service/src/service/api/health/ready_get.rs

```diff
@@ -35,7 +40,22 @@ pub(crate) type AllResponses = response! {
 /// * 500 Server Error - If anything within this function fails unexpectedly. (Possible but unlikely)
 /// * 503 Service Unavailable - Service is not ready, do not send other requests.
 #[allow(clippy::unused_async)]
-pub(crate) async fn endpoint() -> AllResponses {
+pub(crate) async fn endpoint(state: Data<&Arc<State>>) -> AllResponses {
     // otherwise everything seems to be A-OK
-    T204(NoContent)
+    match state.event_db.schema_version_check().await {
```

Comment:
Seems like it makes sense to extract this code to the separate function, because it is used more than one place

Reply:
I agree. At a first attempt, I realize that I need to learn better how to make use of poem-extensions to move this into a cleaner way of handling 5xx responses. If you don't mind, @Mr-Leshiy , this could be done in a separate PR.